### PR TITLE
Fixes number overflowing cell boundaries

### PR DESCRIPTION
**Pull request by davidfig**
Fixes this:

image

### CODE REVIEW

File: quadratic-client/src/app/web-workers/renderWebWorker/worker/cellsLabel/CellLabel.ts [hunk 1]
```diff
@@ -528,7 +529,8 @@ export class CellLabel {
 
     // This attempts to reduce the decimal precision to ensure the number fits
     // within the cell. If it doesn't, it shows the pounds
-    if (this.number && this.textWidth > this.AABB.width && this.displayedText) {
+    const maxWidth = this.AABB.width - CURSOR_BORDER_SIZE * 2;
```
Comment (ddimaria) [inline quadratic-client/src/app/web-workers/renderWebWorker/worker/cellsLabel/CellLabel.ts:532]: The changes in this file seem to be the only material changes.  Can you make a comment in the code for the significance of the the `* 2` portion?

File: quadratic-client/src/app/web-workers/renderWebWorker/worker/cellsLabel/CellLabel.ts [hunk 2]
```diff
@@ -543,7 +545,8 @@ export class CellLabel {
 
         text = result.number;
         this.updateText(labelMeshes, text);
-      } while (this.textWidth > this.AABB.width && digits >= 0 && infinityProtection++ < 1000);
+        console.log(this.textWidth, this.AABB.width - CURSOR_BORDER_SIZE * 2);
```
Comment (ddimaria) [inline quadratic-client/src/app/web-workers/renderWebWorker/worker/cellsLabel/CellLabel.ts:548]: Remove this log?

**Review by ddimaria — APPROVED** [review]

**Review by davidkircos — CHANGES_REQUESTED** [review]
There is a bug when rendering `111111111111`

https://github.com/user-attachments/assets/c92fe517-1b76-4172-8583-605947c8a641

**Comment by davidfig** [comment]
This has a better fix in #1668. Thanks @AyushAgrawal-A2!

**Pull request closed without merging**